### PR TITLE
Stats revamp v2 latest post summary card

### LINKED ISSUE #16424: [android] Stats:  Latest Post Summary Card
Existing card, needs update

### PR DESCRIPTION
**Pull request by ravishanker**
This PR updates Latest Post Summary card to new design as shown below
- Adds Latest Post with featured image (if set)
- Displays stats in three columns (and removes graph)

TODO:
- [x] [Corresponding FluxC PR](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2373) needs to be merged before merging this

Fixes #16424

| Current | New Design | Dark Theme |
|---|---|---|
|Screenshot_20220429_151424|Screenshot_20220429_151749|Screenshot_20220429_151809 |

To test:

Setup:

- Go to App Settings (Tap on Avatar at the top right hand corner on My Site -> Find App Settings)
- Select Debug Settings
- Find StatsRevampV2FeatureConfig under Features in development enable it and restart app

Test 1:

- Launch app
- Go to stats either using quick links or menu
- Ensure that it opens in Insights tab otherwise switch to Insights tab
- Verify that the new Latest Post Summary card is visible and matches the new designs above
- Ensure that featured image is displayed (if a feature image is set)
- View More on the top right corner is visible if there's any data on Views, Likes or Comments
- If data is not available it will display a Share action or Create Post if no latest post as before
- _If the Latest Post Summary card is not present.  Go to stats management (either through ⚙️ menu at the top right hand corner or Add new stats card at the bottom) and add the card and save then return to Insights._

Test 2:

- Turn on the dark theme on your device
- Check the card is displayed as in third image above

Test 3:

- Disable StatsRevampV2FeatureConfig and restart app
- Return to Stats and make sure Most Popular Time card is displayed as in first image above as before.

## Regression Notes
1. Potential unintended areas of impact
None.  Only existing Latest Post Summary card will be updated to new design with additions of a featured image

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing

3. What automated tests I added (or what prevented me from doing so)
Updated unit tests

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/src/main/res/layout/stats_quick_scan_item.xml [hunk 1]
```diff
@@ -95,4 +99,59 @@
...
+            android:text="@string/unknown"
+            tools:text="@string/stats_comments"/>
+
+        <TextView
+            android:id="@+id/third_value"
+            style="@style/StatsBlockQuickScanValue"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:text="@string/unknown"
+            tools:text="1"/>
```
Comment (irfano) [inline WordPress/src/main/res/layout/stats_quick_scan_item.xml:146]: I see you added a third column to `stats_quick_scan_item.xml`. That's a good way to use this layout with the old and new designs. But there are some differences between the old layout and the new design in Figma:
- In the new design, the whole layout has 90dp end margin.
- Value text size is 24sp in the old design and 34sp in the new design.

How can we handle this? Maybe a new layout for stats revamp v2 could be a better solution.

image

File: WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/lists/sections/insights/usecases/LatestPostSummaryMapper.kt [hunk 2]
```diff
@@ -22,6 +24,24 @@ class LatestPostSummaryMapper
     private val statsDateFormatter: StatsDateFormatter,
     private val statsUtils: StatsUtils
 ) {
+    fun buildLatestPostItem(
+        model: InsightsLatestPostModel?
+    ): ListItemWithImage {
+        if (model == null) {
+            return ListItemWithImage(title = resourceProvider.getString(string.stats_insights_latest_post_empty))
```
Comment (irfano) [inline WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/lists/sections/insights/usecases/LatestPostSummaryMapper.kt]: We can use only text for the empty card, without the image gap.

|PR version|Figma version|
|-|-|
|image|image|

File: WordPress/src/main/res/layout/stats_block_title_with_more_item.xml [hunk 3]
```diff
@@ -10,7 +10,7 @@
 
     <com.google.android.material.textview.MaterialTextView
         android:id="@+id/text"
-        style="@style/TextAppearance.LoginFlow.Subtitle1"
+        style="@style/StatsBlockTitle"
```
Comment (irfano) [inline WordPress/src/main/res/layout/stats_block_title_with_more_item.xml:13]: It looks like the text size is `20sp` in the design, but  `16sp` in the `StatsBlockTitle`. I think we can use a different style for Stats Revamp v2.
We can do this in another PR.

**Review by irfano — CHANGES_REQUESTED** [review]

File: WordPress/src/main/res/layout/stats_quick_scan_item.xml (hunk 1 above)
Comment (ravishanker) [reply WordPress/src/main/res/layout/stats_quick_scan_item.xml:146]: added margin.  thank you

File: WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/lists/sections/insights/usecases/LatestPostSummaryMapper.kt (hunk 2 above)
Comment (ravishanker) [reply WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/lists/sections/insights/usecases/LatestPostSummaryMapper.kt]: Good point, was going to do as part empty states.  Fixed it anyhow by hiding image when empty

File: WordPress/src/main/res/layout/stats_block_title_with_more_item.xml (hunk 3 above)
Comment (ravishanker) [reply WordPress/src/main/res/layout/stats_block_title_with_more_item.xml:13]: Yes, we can do this as part styles pr

**Comment by AliSoftware** [comment]
I'm doing 19.8 code freeze today, so migrating this PR's milestone to 19.9.

(PS: Please ping me to let me know if you _really_ needed this in 19.8, as this would need a rebase that would impact release notes, CfT, and beta wrangling)

**Comment by irfano** [comment]
I've merged the corresponding [FluxC PR](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2373) before this PR. That caused failing unit tests because of the new parameter, `featuredImageUrl`.
A fix is added to the `release/19.8` branch in this [PR](https://github.com/wordpress-mobile/WordPress-Android/pull/16455).

File: WordPress/src/main/res/layout/stats_quick_scan_item.xml (hunk 1 above)
Comment (irfano) [reply WordPress/src/main/res/layout/stats_quick_scan_item.xml:144]: Now we have a new problem. :sweat_smile:

Screenshot_20220502_163541

**Review by irfano — CHANGES_REQUESTED** [review]
Now we have a new problem. 😅

Screenshot_20220502_163541

**Comment by ravishanker** [comment]
> Now we have a new problem. 😅

🥹  Looks like there's a mistake in the design actually.  If you look at [iOS designs and their implementation](https://github.com/wordpress-mobile/WordPress-iOS/pull/18476), there's no padding at the end.  Also, adding a padding is causing issues with other languages

**Review by irfano — APPROVED** [review]
> 🥹 Looks like there's a mistake in the design actually. If you look at [iOS designs and their implementation](https://github.com/wordpress-mobile/WordPress-iOS/pull/18476), there's no padding at the end. Also, adding a padding is causing issues with other languages

Oh! That's right! Now I also checked Figma file, it must be a design mistake.
Approved, LGTM! 👍🏻

**Pull request merged**